docs/plugin-writer/tasks/publish.rst

```diff
@@ -2,8 +2,12 @@ Publish
 =======
 
 
-TODO Publication
-https://pulp.plan.io/issues/3298
+A publication contains metadata and artifacts associated with content
+contained within a RepositoryVersion.
+
+Using a context manager is highly encouraged.  On context exit, the complete attribute is set
```

Comment:
naming the context manager, and maybe adding a link to the plugin-api docs would be helpful.